go/interpreter/lfvm/instructions.go

```diff
@@ -1081,22 +1070,25 @@ func genericCall(c *context, kind tosca.CallKind) {
 	// and not doing it would be identified by the replay tool as an error.
 	toAddr := tosca.Address(addr.Bytes20())
 
-	// Compute and charge gas price for call
-	arg_memory_size, err := neededMemorySize(c, inOffset, inSize)
-	if err != nil {
+	if checkSizeOffsetUint64Overflow(inOffset, inSize) != nil ||
+		checkSizeOffsetUint64Overflow(retOffset, retSize) != nil {
+		c.signalError()
 		return
 	}
-	ret_memory_size, err := neededMemorySize(c, retOffset, retSize)
+
+	// Get arguments from the memory.
+	args, err := c.memory.GetSliceWithCapacityAndGas(inOffset.Uint64(), inSize.Uint64(), c)
 	if err != nil {
+		c.signalError()
 		return
 	}
-
-	needed_memory_size := arg_memory_size
-	if ret_memory_size > arg_memory_size {
-		needed_memory_size = ret_memory_size
+	output, err := c.memory.GetSliceWithCapacityAndGas(retOffset.Uint64(), retSize.Uint64(), c)
```

Comment:
I assume `GetSliceWithCapacityAndGas` will be renamed by some other PR into something simpler like `getSlice`.

Reply:
exactly